types/knockout.mapping/index.d.ts

```diff
@@ -15,13 +15,23 @@ export = self;
 
 
 declare global {
-    // the idea behind this is good, but doesn't work as intended. This will make object properties T | Observable<T>, 
-    //and you will have to put the correct Type in each property to use them, what defeats the purpose. Besides, this gives  
-    //RangeError: Maximum call stack size exceeded in TSC when used in all the mapping signatures. Maybe this can be used with TS 2.8 and conditional typing
+
+    type Primitives = string | number | boolean;
```

Comment:
Is this the full list of primitives supported by knockout, or should this also include `bigint | symbol`?

Reply:
Also, I would rather see the types for `Primitives` inlined below rather than have its own declaration, or defined *outside* of the `declare global`. Adding something like `Primitives` to the global scope is bound to end up with a "duplicate identifier" conflict at some point.

Reply:
Added support for this scenario, and made necessary changes.